pkg/virt-handler/non-root.go

```diff
@@ -120,13 +120,18 @@ func getTapDevices(vmi *v1.VirtualMachineInstance) []string {
 		_, isMacvtapNetwork := macvtap[net.Name]
 		if podInterfaceName, exists := networkNameScheme[net.Name]; isMacvtapNetwork && exists {
 			tapDevices = append(tapDevices, podInterfaceName)
+		} else if !exists {
```

Comment:
This is confusing, because we reach this "else" even if this is not a macvtap binding.
In other words, we may fail not because macvtap related problems.

Can you make sure the error is raised only for macvtap interfaces/networks?

Reply:
My point is that you will raise an error that is not related to macvtap, i.e. expose a bug which has nothing to do with macvtap. I am just thinking of the poor one that needs to figure this out.

I do not think it is much truble to only handler macvtap items (i.e. lookup the name only if this is a macvtap binding).

Reply:
> My point is that you will raise an error that is not related to macvtap, i.e. expose a bug which has nothing to do with macvtap. I am just thinking of the poor one that needs to figure this out.
> 
> I do not think it is much truble to only handle macvtap items (i.e. lookup the name only if this is a macvtap binding).

It isn't about the trouble. I agree doing that is very easy. 

My point is only if this happens, how can I trust anything this pod does ? If this weren't virt-handler I'll panic the pod and let it re-schedule. ... half tempted to do it anyway.

Reply:
Bugs pop up from time to time. It will be unfortunate that a macvtap handler blow up for scenarios which do not include macvtap at all.
I do not think you really intended to do this intentionally.

Reply:
I agree with Ed, we should isolate this to macvtap.

Reply:
Done.